src/tests/dotnet-counters/JSONExporterTests.cs

```diff
@@ -0,0 +1,107 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.Collections.Generic;
+using Xunit;
+using Microsoft.Diagnostics.Tools.Counters.Exporters;
+using Newtonsoft.Json;
+
+namespace DotnetCounters.UnitTests
+{
+    /// <summary>
+    /// These test the some of the known providers that we provide as a default configuration for customers to use.
+    /// </summary>
+    public class JSONExporterTests
+    {
+        [Fact]
+        public void IncrementingCounterTest()
+        {
+            string fileName = "IncrementingCounterTest.json";
+            JSONExporter exporter = new JSONExporter(fileName, "myProcess.exe");
+            exporter.Initialize();
+            for (int i = 0; i < 10; i++)
+            {
+                exporter.CounterPayloadReceived("myProvider", TestHelpers.GenerateCounterPayload(true, "incrementingCounterOne", 1.0, 1, "Incrementing Counter One"), false);
+            }
+            exporter.Stop();
+
+            Assert.True(File.Exists(fileName));
+            using (StreamReader r = new StreamReader(fileName))
+            {
+                string json = r.ReadToEnd();
+                JSONCounterTrace counterTrace = JsonConvert.DeserializeObject<JSONCounterTrace>(json);
+
+                Assert.Equal("myProcess.exe", counterTrace.targetProcess);
+                foreach (JSONCounterPayload payload in counterTrace.events)
```

Comment:
Same issue as the csv case, the test isn't validating the number of entries and each entry is identical so lots of room for bugs to lurk.